packages/vm/src/vm.rs

```diff
@@ -123,13 +100,22 @@ where
 
     fn with_context_data<C, R>(&self, callback: C) -> R
     where
-        C: FnOnce(&ContextData) -> R,
+        C: FnOnce(&ContextData<Q>) -> R,
     {
         let guard = self.data.as_ref().read().unwrap();
         let context_data = guard.borrow();
         callback(context_data)
     }
 
+    // fn with_context_data_mut<C, R>(&self, callback: C) -> R
```

Comment:
Remove?

Reply:
yes, but it is fundamental function that is likely to be used in the future so I think we better just commented it